overlord/snapstate/snapmgr.go

```diff
@@ -420,6 +431,10 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 			newSeq = append(newSeq, si)
 		}
 		snapst.Sequence = newSeq
+		// update Current if it just got removed
+		if snapst.Current == ss.Revision {
```

Comment:
actually we should error inside here if snapst.Active  (probably the same in the other branch where we unset current)